### PR TITLE
Added new device - Exagrid

### PR DESCRIPTION
**Pull request by nsn-amagruder**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: mibs/exagrid/* [hunk 1]
```diff
@@ -0,0 +1,353 @@
+EXAGRID-MIB DEFINITIONS ::= BEGIN
```
Comment (PipoCanaja) [inline mibs/exagrid/*]: Please rename the MIB file to 'EXAGRID-MIB'

File: tests/data/exagridos.json [hunk 2]
```diff
@@ -0,0 +1,2037 @@
...
+        },
+        "poller": {
+            "devices": [
+                {
+                    "sysName": "<private>",
+                    "sysObjectID": ".1.3.6.1.4.1.14941.3.1.840",
+                    "sysDescr": "ExaGrid EX40000E-SEC, 4.8.1.1209.P60",
+                    "sysContact": "<private>",
+                    "version": null,
+                    "hardware": null,
```
Comment (PipoCanaja) [inline tests/data/exagridos.json]: Seems that version is not collected, nor HW. You can probably check in the snmpwalk data from the device, and get them from an OID, if you have no good way to extract them from SysDescr.

Comment (nsn-amagruder) [reply tests/data/exagridos.json]: snmpwalk doesn't show there version or hw in another field than the sysDescr field.  If there is a way to separate them out in exagrid.inc.php, it is beyond me.

RFC1213-MIB::sysDescr.0 = STRING: "ExaGrid EX40000E-SEC, 4.8.1.1209.P60"

File: mibs/exagrid/* (hunk 1 above)
Comment (nsn-amagruder) [reply mibs/exagrid/*]: Ya, this was a bad copy.  Will upload.

File: tests/data/exagridos.json (hunk 2 above)
Comment (PipoCanaja) [reply tests/data/exagridos.json]: OK. Then you probably need to adapt ``` includes/polling/os/exagrid.inc.php ``` to collect the version from here.

**Comment by PipoCanaja** [comment]
When your changes (if necessary, for version/hardware etc) will be pushed, you can check Travis result and correct whatever issue is found. Right now, the test data is not matching the snmprec, which makes Travis fail.

File: tests/data/exagridos.json (hunk 2 above)
Comment (nsn-amagruder) [reply tests/data/exagridos.json]: If you can tell me how to do that, great.  If not, just close this.  Just trying to help contribute.  It shows the version and hardware version in my setup.

Comment (PipoCanaja) [reply tests/data/exagridos.json]: you have code in ``` includes/polling/os/exagrid.inc.php  ```to fill the $version, which tries to parse sysDescr to extract it. But the Regexp does not match. 

So either you remove ``` includes/polling/os/exagrid.inc.php ``` (it does not do anything right now in your environnement) or patch it to actually match the bold part of sysDescr : 

ExaGrid EX40000E-SEC, **4.8.1.1209.P60**

Something like the code below should extract both Hardware and version to the proper variables. 
```
preg_match('/^ExaGrid ([^,]*), (.*)$/', $device['sysDescr'], $data);
$version = $data[2];
$hardware = $data[1];
```

Comment (nsn-amagruder) [reply tests/data/exagridos.json]: Thanks,  This is not my specialty.  Took me some time to get adding the device figured out.  I tried removing the php file all together and it didn't match.  Updated to what you had and renamed the file to exagridos.php and it is matching  now.  Will try to upload shortly.  

#### Load poller module os ####
Hardware: EX40000E-SEC
Version: 4.8.1.1209.P60
Features: 
Serial:

**Comment by nsn-amagruder** [comment]
Travis CI showing disk don't match.  I think this may be the way their disk is reported in SNMP.  This is a backup appliance and they partition the disk differently.  Thoughts?

snmprec file
1.3.6.1.2.1.25.2.3.1.3.35|4|/dev/shm
1.3.6.1.2.1.25.2.3.1.3.39|4|/home1

json doesn't report disk 35.
                    "storage_index": "31",
                    "storage_type": "hrStorageFixedDisk",
                    "storage_descr": "/",

                    "storage_index": "39",
                    "storage_type": "hrStorageFixedDisk",
                    "storage_descr": "/home1",



OS exagridos: Discovered storage data does not match that found in tests/data/exagridos.json
Failed asserting that two arrays are equal.

**Comment by PipoCanaja** [comment]
Concerning the tests, I can see 2 explanations : 
- You have some options in your config.php to skip some of the volumes
- You run the test on a different snmprec than the one attached to this PR. 

Anyway, I can run it on my dev machine. We'll see how it goes.

**Comment by nsn-amagruder** [comment]
> Concerning the tests, I can see 2 explanations :
> 
> * You have some options in your config.php to skip some of the volumes
> * You run the test on a different snmprec than the one attached to this PR.
> 
> Anyway, I can run it on my dev machine. We'll see how it goes.

I do exclude some volumes in my config.php.  Should I add the same excludes to the config.text.php file?

**Comment by PipoCanaja** [comment]
In fact, you need to do the dev on a "pure" LibreNMS instance. If you have config that impacts discovery or poller (which you most probably have on a productive machine) your tests will reflect this.
And when the automatic tests are run in a "pure" LibreNMS instance by travis, they fail the equality test. 

I did run the tests again in a dev machine, that should be better.

**Comment by nsn-amagruder** [comment]
Appreciate your help.  First try for me.  Will keep this in mind for the future.

**Comment by PipoCanaja** [comment]
We all had a first time :) If you can, it is definitely easier to have a dedicated dev box (VM) for playing with GIT and tests. 
You can play with different branches without impacting your productive monitoring, and keep your productive instance in sync with auto-updates.

**Comment by PipoCanaja** [comment]
@nsn-amagruder There we are. Checks OK, ready for merge.

**Comment by PipoCanaja** [comment]
You'll need to clean your changes in the LibreNMS directory before running daily.php. If not, the auto-upgrade will complain.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-55-release-changelog-august-2019/9428/1

**Pull request merged**